LLama/Native/Load/NativeLibraryConfig.cs

```diff
@@ -0,0 +1,605 @@
+        public NativeLibraryConfigContainer WithSearchDirectories(IEnumerable<string> directories)
+        {
+            foreach (var config in _configs)
+            {
+                config.WithSearchDirectories(directories);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Add self-defined search directories. Note that the file stucture of the added 
+        /// directories must be the same as the default directory. Besides, the directory 
+        /// won't be used recursively.
+        /// </summary>
+        /// <param name="directory"></param>
+        /// <returns></returns>
+        public NativeLibraryConfigContainer WithSearchDirectory(string directory)
+        {
+            foreach (var config in _configs)
+            {
+                config.WithSearchDirectory(directory);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Set the policy which decides how to select the desired native libraries and order them by priority. 
+        /// By default we use <see cref="DefaultNativeLibrarySelectingPolicy"/>.
+        /// </summary>
+        /// <param name="policy"></param>
+        /// <returns></returns>
+        public NativeLibraryConfigContainer WithSelectingPolicy(INativeLibrarySelectingPolicy policy)
+        {
+            foreach (var config in _configs)
+            {
+                config.WithSelectingPolicy(policy);
+            }
+            return this;
+        }
+#endif
+
+        /// <summary>
+        /// Set the log callback that will be used for all llama.cpp log messages
+        /// </summary>
+        /// <param name="callback"></param>
+        /// <exception cref="NotImplementedException"></exception>
+        public NativeLibraryConfigContainer WithLogCallback(NativeLogConfig.LLamaLogCallback? callback)
+        {
+            foreach (var config in _configs)
+            {
+                config.WithLogCallback(callback);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Set the log callback that will be used for all llama.cpp log messages
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <exception cref="NotImplementedException"></exception>
+        public NativeLibraryConfigContainer WithLogCallback(ILogger? logger)
+        {
+            foreach (var config in _configs)
+            {
+                config.WithLogCallback(logger);
+            }
+            return this;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Try to load the native library with the current configurations, 
+        /// but do not actually set it to <see cref="NativeApi"/>.
+        /// 
+        /// You can still modify the configuration after this calling but only before any call from <see cref="NativeApi"/>.
+        /// </summary>
+        /// <returns>Whether the running is successful.</returns>
+        public bool DryRun()
```

Comment:
This returns if it was succesful, but doesn't return _which_ native library was selected. That seems a bit odd for a dry run?

Reply:
You're right! Maybe the following API will be good

```cs
public bool DryRun(out var ILibrary?)
```